### DISCUSSION TITLE
Installation on RPi with Ubuntu MATE

### DISCUSSION THREAD

**Discussion by domy86** [Arkq/bluez-alsa#448] [Q&A]
Hi,

Your repository regarding A2DP seems to be the most recent one out there so I would like to ask you for an advice.
I have installed a fresh Ubuntu MATE 20.04.2 on my RPi 4 and here are the packages related to Alsa, Pulse and Bluetooth that I have found. Sound is going through PulseAudio but I've gotten some components for both BlueZ and Alsa. OK, I know that Pulse sits on Alsa but just want to show you.
Is it possible to install your repository on my distro without causing some other issues? Please be informed that I use RPi as a multimedia player and there is one sink more active.

```
domy@domy-pi:~$ apt list --installed *alsa*
Listing... Done
alsa-base/focal,now 1.0.25+dfsg-0ubuntu5 all [installed]
alsa-topology-conf/focal,now 1.2.2-1 all [installed]
alsa-ucm-conf/focal-updates,now 1.2.2-1ubuntu0.7 all [installed,automatic]
alsa-utils/focal-updates,now 1.2.2-1ubuntu2.1 arm64 [installed]
gstreamer1.0-alsa/focal-updates,focal-security,now 1.16.2-4ubuntu0.1 arm64 [installed]
```

```
domy@domy-pi:~$ apt list --installed *pulse*
Listing... Done
gstreamer1.0-pulseaudio/focal-updates,focal-security,now 1.16.2-1ubuntu2.1 arm64 [installed]
libcanberra-pulse/focal,now 0.30-7ubuntu1 arm64 [installed]
libpulse-dev/focal-updates,now 1:13.99.1-1ubuntu3.10 arm64 [installed]
libpulse-mainloop-glib0/focal-updates,now 1:13.99.1-1ubuntu3.10 arm64 [installed]
libpulse0/focal-updates,now 1:13.99.1-1ubuntu3.10 arm64 [installed]
libpulsedsp/focal-updates,now 1:13.99.1-1ubuntu3.10 arm64 [installed]
pulseaudio-module-bluetooth/focal-updates,now 1:13.99.1-1ubuntu3.10 arm64 [installed]
pulseaudio-module-gsettings/focal-updates,now 1:13.99.1-1ubuntu3.10 arm64 [installed,automatic]
pulseaudio-module-zeroconf/focal-updates,now 1:13.99.1-1ubuntu3.10 arm64 [installed,automatic]
pulseaudio-utils/focal-updates,now 1:13.99.1-1ubuntu3.10 arm64 [installed]
pulseaudio/focal-updates,now 1:13.99.1-1ubuntu3.10 arm64 [installed]
pulsemixer/focal,now 1.5.0-1 all [installed]
```

```
domy@domy-pi:~$ apt list --installed *blue*
Listing... Done
blueman/focal-updates,focal-security,now 2.1.2-1ubuntu0.2 arm64 [installed]
bluez-cups/focal-updates,now 5.53-0ubuntu3.1 arm64 [installed]
bluez-obexd/focal-updates,now 5.53-0ubuntu3.1 arm64 [installed]
bluez/focal-updates,now 5.53-0ubuntu3.1 arm64 [installed]
libbluetooth3/focal-updates,now 5.53-0ubuntu3.1 arm64 [installed]
pi-bluetooth/focal-updates,now 0.1.15ubuntu0~20.04.1 arm64 [installed]
pulseaudio-module-bluetooth/focal-updates,now 1:13.99.1-1ubuntu3.10 arm64 [installed]
```

Thank you in advance and I wish you all the best!

**Comment by borine** [comment]
> Is it possible to install your repository on my distro without causing some other issues?

Unlikely. MATE desktop is built on pulseaudio for all audio management, and many of its core applications require pulseaudio for full functionality. It's not clear why you would want to install bluez-alsa when you already have bluetooth audio support in pulseaudio? If you have some specific reason in mind, it might be possible to give a more specific answer.

**Reply by domy86** [reply]
I would like to use RPi as receiver (sink) for streaming music from the phone. I think I can not get a better codec (AAC or aptX) with PulseAudio but maybe I'm wrong.

**Reply by borine** [reply]
ah, I see.

Pulseaudio has now implemented aptx, aac and ldac (https://gitlab.freedesktop.org/pulseaudio/pulseaudio/-/merge_requests/440/), but that code is not in any release yet, I think. So your first option is to build pulse yourself from latest code.

There is also https://github.com/EHfive/pulseaudio-modules-bt which can be used to add extra bluetooth codec support to pulseaudio. So your second option is to build and install that module to add the extra codecs to your existing pulseaudio installation.

Or your third option would be to disable the pulseaudio bluetooth modules and build bluez-alsa. That would remove bluetooth support from many, maybe all, your desktop applications but would allow you to create a bluetooth sink using bluealsa-aplay .

**Reply by domy86** [reply]
I understand, thanks a lot! I will switch to BlueAlsa if something goes wrong with the first two options.
